.gitlab/deps_fetch/deps_fetch.yml

```diff
@@ -66,6 +66,7 @@ go_tools_deps:
   script:
     - if [ -f modcache_tools.tar.xz  ]; then exit 0; fi
     - inv -e install-tools
+    - go install github.com/DataDog/datadog-packages/cmd/datadog-package@latest # This is needed to provide cached package for the agent_oci job to mitigate #incident-30779. This can be removed when datadog-packages is stored internally as a binary
```

Comment:
I think this fails without a GitHub -> Gitlab git redirection, like the OCI jobs do, since this targets a private repo.

I think we also want to note that this is only a partial mitigation, because the results of the `go_tools_deps` jobs are also cached. Therefore, if `datadog-packages` updates its latest without updating the cache key components here, we'll be out of date.